File: tests/fixtures/selenium-chaos-suite-csharp/src/Flows/CheckoutFlow.cs
using OpenQA.Selenium;
using SeleniumChaosSuite.Models;
using SeleniumChaosSuite.Pages;

namespace SeleniumChaosSuite.Flows;

public class CheckoutFlow
{
    private readonly LoginPage _loginPage;
    private readonly ProductsPage _productsPage;
    private readonly CartPage _cartPage;
    private readonly CheckoutPage _checkoutPage;

    public CheckoutFlow(IWebDriver driver)
    {
        _loginPage = new LoginPage(driver);
        _productsPage = new ProductsPage(driver);
        _cartPage = new CartPage(driver);
        _checkoutPage = new CheckoutPage(driver);
    }

    public async Task RunHappyPathCheckout()
    {
        await _loginPage.LoginWith(Users.Standard.Username, Users.Standard.Password);
        await _productsPage.GetPageTitle();
        await _productsPage.AddBackpackToCart();
        await _productsPage.AddBikeLightToCart();
        await _productsPage.GoToCartFromHeader();
        await _cartPage.GetCartTitle();
        await _cartPage.GoToCheckout();
        await _checkoutPage.FillCheckoutInfo(Users.Standard.FirstName, Users.Standard.LastName, Users.Standard.Zip);
        await _checkoutPage.GetSummaryInfo();
        await _checkoutPage.GetTotalLabel();
        await _checkoutPage.FinishCheckout();
    }

    public async Task RunAddRemoveReaddCheckout()
    {
        await _loginPage.LoginWith(Users.Standard.Username, Users.Standard.Password);
        await _productsPage.AddBackpackToCart();
        await _productsPage.GoToCartFromHeader();
        await _cartPage.RemoveFirstItem();
        await _cartPage.ContinueShopping();
        await _productsPage.AddAllItemsOneByOne();
        await _productsPage.GoToCartFromHeader();
        await _cartPage.GoToCheckout();
        await _checkoutPage.FillCheckoutInfo("A", "B", "00000");
        await _checkoutPage.FinishCheckout();
    }
}

File: tests/fixtures/selenium-chaos-suite/src/flows/CheckoutFlow.ts
import { LoginPage } from "../pages/LoginPage";
import { ProductsPage } from "../pages/ProductsPage";
import { CartPage } from "../pages/CartPage";
import { CheckoutPage } from "../pages/CheckoutPage";
import { users } from "../data/users";

// Intentionally procedural and repetitive flow class.
export class CheckoutFlow {
  loginPage = new LoginPage();
  productsPage = new ProductsPage();
  cartPage = new CartPage();
  checkoutPage = new CheckoutPage();

  async runHappyPathCheckout() {
    await this.loginPage.loginWith(users.standard.username, users.standard.password);
    await this.productsPage.assertInventoryPageLoaded();
    await this.productsPage.addBackpackToCart();
    await this.productsPage.addBikeLightToCart();
    await this.productsPage.goToCartFromHeader();
    await this.cartPage.assertCartPageLoaded();
    await this.cartPage.goToCheckout();
    await this.checkoutPage.fillCheckoutInfo(users.standard.firstName, users.standard.lastName, users.standard.zip);
    await this.checkoutPage.assertSummaryContainsPayment();
    await this.checkoutPage.assertTotalLooksValid();
    await this.checkoutPage.finishCheckout();
  }

  async runAddRemoveReaddCheckout() {
    await this.loginPage.loginWith(users.standard.username, users.standard.password);
    await this.productsPage.addBackpackToCart();
    await this.productsPage.goToCartFromHeader();
    await this.cartPage.removeFirstItem();
    await this.cartPage.continueShopping();
    await this.productsPage.addAllItemsOneByOne();
    await this.productsPage.goToCartFromHeader();
    await this.cartPage.goToCheckout();
    await this.checkoutPage.fillCheckoutInfo("A", "B", "00000");
    await this.checkoutPage.finishCheckout();
  }
}
